test/zdtm.py

```diff
@@ -112,7 +112,7 @@ def add_to_output(path):
         return
 
     output_path = os.path.join(report_dir, "output")
-    with open(path, "r") as fdi, open(output_path, "a") as fdo:
+    with builtins.open(path, "r") as fdi, builtins.open(output_path, "a") as fdo:
```

Comment:
Do we still need this module? I remember we added it to make zdtm.py compatible with py2 and py3. Do we still want to be compatible with py2?

I don't like the idea of this change. At some point, we will need to stop using this module and other py2-compatibility hacks.

Reply:
> Do we still want to be compatible with py2?

The CentOS 7 CI tests are still running with Python 2.

> I don't like the idea of this change. At some point, we will need to stop using this module and other py2-compatibility hacks.

Should I just revert this change or drop py2-compatibility hacks?

@adrianreber What do you think about the Python 2/3 support for zdtm?

Reply:
> > Do we still want to be compatible with py2?
> 
> The CentOS 7 CI tests are still running with Python 2.
> 
> > I don't like the idea of this change. At some point, we will need to stop using this module and other py2-compatibility hacks.
> 
> Should I just revert this change or drop py2-compatibility hacks?
> 
> @adrianreber What do you think about the Python 2/3 support for zdtm?

From my point of view CentOS/RHEL7 is so old that we do not have to support Python 2 any more. I proposed it at some point but @Snorch said for their use of CRIU they would prefer if Python 2 is still supported to be able to run tests.

Reply:
From the first look I failed to run zdtm on VZ7 via python3:

```
make -j5 PYTHON=python3
[root@silo criu]# grep unicode /root/build/criu/test/pycriu/rpc_pb2.py
      has_default_value=False, default_value=unicode("", "utf-8"),
[root@silo criu]# python3 ./test/zdtm.py run -t zdtm/static/env00 --keep-going --ignore-taint
Traceback (most recent call last):
  File "./test/zdtm.py", line 26, in <module>
    import pycriu as crpc
  File "/root/build/criu/test/pycriu/__init__.py", line 1, in <module>
    from . import rpc_pb2 as rpc
  File "/root/build/criu/test/pycriu/rpc_pb2.py", line 184, in <module>
    has_default_value=False, default_value=unicode("", "utf-8"),
NameError: name 'unicode' is not defined
```
even after `pip3 install protobuf` it still somehow generates python2 code in pycriu... I'm not too good in makefiles so probably I'm doing something wrong.

When run make criu on fedora I see `b"".decode('utf-8')` instead of `unicode` there, which helps it to run on python3 but I don't know why.

Reply:
I think zdtm restarts zdtm. So you have to change the shebang to python3 to make sure it does not use python2

Reply:
I mean that If even I remove python symlink and python2 binary completely, protobuf (when make `PYTHON=python3 -j5`) still generates code in lib/py/rpc_pb2.py which does not run on python3.

Reply:
@Snorch We still have many `#!/usr/bin/python` shebang lines in the tests. The following line makes the tests work in CI with python3:

https://github.com/checkpoint-restore/criu/blob/8b0fabc92dfd5bce25e9e2d1f978100ece5ef254/scripts/ci/run-ci-tests.sh#L61

Reply:
> still generates code in lib/py/rpc_pb2.py which does not run on python3.

This comment explains why:
https://github.com/checkpoint-restore/criu/blob/8b0fabc92dfd5bce25e9e2d1f978100ece5ef254/lib/py/images/Makefile#L6

Reply:
> NameError: name 'unicode' is not defined

I cannot replicate this error on Fedora 35. The file generated with `make PYTHON=python3` doesn't contain `unicode`.
This file is generated with `protoc -I=images/ --python_out=lib/py images/rpc.proto`

Reply:
``` ln -sf /usr/bin/python3 /usr/bin/python ```
does not help either, on Centos7 based environment I still get unicode in there...